### PR TITLE
Add WebRTC Simulcast Support

### PR DESCRIPTION
**Pull request by Sean-Der**
### Description
WebRTC allows broadcasters to upload multiple streams of different quality. OBS users could upload ‘high’, ‘med’, and ‘low’ streams themselves. 

### Motivation and Context

* Higher Quality - Re-encoding video server side causes quality loss
* Lower Latency - Reduces the amount of time before video gets to viewers
* Less complicated servers - Users find it difficult to setup RTMP->HLS with transcodes. This removes the complexity. 

### How Has This Been Tested?
Tested against https://github.com/glimesh/broadcast-box

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by notr1ch** [comment]
I started to review this but it seems like this is still at the proof of concept stage, there are too many hard coded assumptions and the deps point to your own repo. I don't think this was ready to be PR'ed.

**Comment by Sean-Der** [comment]
Hey @notr1ch appreciate the review! I put the PR up to get feedback about the right way to do this. Instead of hardcoding these values what do you think would be best for users?

* UI where people can set the settings per layer (bitrate, resolution...) 
* UI where users can select the amount of layers, but not configure the layers
* UI where you enable Simulcast or disable, but no configuration.
* Something else?

File: UI/window-basic-main-outputs.cpp [hunk 1]
```diff
@@ -1821,10 +1833,18 @@ inline void AdvancedOutput::SetupRecording()
 		tracks = config_get_int(main->Config(), "AdvOut", "TrackIndex");
 
 	if (useStreamEncoder) {
-		obs_output_set_video_encoder(fileOutput, videoStreaming);
+		// Sean-Der
+		obs_encoder_set_scaled_size(videoStreaming[0], 640, 480);
+		obs_encoder_set_scaled_size(videoStreaming[1], 800, 600);
+		obs_encoder_set_scaled_size(videoStreaming[2], 1024, 768);
```
[the remark this answers is not in the document]
Comment (Sean-Der) [reply UI/window-basic-main-outputs.cpp:1839]: 👍 Responded here https://github.com/obsproject/obs-studio/pull/9165#issuecomment-1611902590

File: UI/window-basic-main-outputs.cpp [hunk 2]
```diff
@@ -1619,13 +1619,25 @@ AdvancedOutput::AdvancedOutput(OBSBasic *main_) : BasicOutputHandler(main_)
...
+
+	videoStreaming[1] = obs_video_encoder_create(streamEncoder,
+						     "advanced_video_stream_2",
+						     streamEncSettings,
+						     nullptr);
+
+	videoStreaming[2] = obs_video_encoder_create(streamEncoder,
+						     "advanced_video_stream_3",
+						     streamEncSettings,
+						     nullptr);
```
[the remark this answers is not in the document]
Comment (Sean-Der) [reply UI/window-basic-main-outputs.cpp:1635]: Responded here https://github.com/obsproject/obs-studio/pull/9165#issuecomment-1611902590

**Comment by Sean-Der** [comment]
@notr1ch I have multiple companies testing this PR. Can you please re-open so I can continue collaborating with people on this highly requested feature.

What are the rules/requirements for a draft PR. Can you please provide the official guidelines

**Comment by Fenrirthviti** [comment]
Guidelines for when an RFC is appropriate can be found on the RFC repo in the readme:

https://github.com/obsproject/rfcs

**Comment by Sean-Der** [comment]
@Fenrirthviti @notr1ch RFC is [here](https://github.com/obsproject/rfcs/pull/55)

Can you please re-open this PR now? It seems to be an accepted pattern to have a example implementation. [here](https://github.com/obsproject/rfcs/blob/master/text/0049-multiple-video-mixes.md?plain=1#L84) and [here](https://github.com/obsproject/rfcs/pull/54/files#diff-8425107e8b800172fed39a4587d09c8bf9c39272d236d2a0c3b027eaefc59a24R78)

**Comment by Fenrirthviti** [comment]
That kind of pattern isn't exactly accepted, it's just happened. Discussion can take place on the RFC, and collaboration on the implementation can take place on your fork if there is still implementation work to be done. It would likely also be best to wait until the RFC is in a state that the project agrees on first, to avoid unnecessary efforts.

I'd prefer that this remain closed until it's ready for actual review to avoid cluttering the repo. Previous PRs in this ecosystem had a lot of troubleshooting and off-topic chatter that I'd like to avoid as well, as it complicated review having to sift through what were effectively support questions, not development questions.

**Pull request closed without merging**